### DISCUSSION TITLE
Voice Preview Edition Sync

### DISCUSSION THREAD

**Discussion by RemakingEden** [music-assistant/support#3391] [Feature requests and ideas]
Hi,

First off thanks for all the work on Music Assistant. I've been using it for a a while now and the pace of improvements is incredible. 

I've been using squeezeamps so far to power my speakers and synced play. 

Today my voice preview editions arrived and it made me wonder if that will be the better long term playback system. 

As these are home assistant media players the big problem is that they don't support audio sync. I wanted to check if this was on the roadmap or is it something not being considered?

Thanks!

**Comment by OzGav** [comment]
At some point in the future this should be possible

**Reply by RemakingEden** [reply]
Great thanks. I'll start pivoting over to this as music system then!

**Reply by OzGav** [reply]
Just to keep you updated. Kevin (who works at Nabu Casa on esphome and voice) managed to get snapcast working within esphome. This is an enabler for fully synced playback on the VPE. No timeline from here but just to let you know some progress has been made.

**Reply by RemakingEden** [reply]
Ah thanks, I really appreciate the update @OzGav

**Reply by maxi1134** [reply]
Any progress a few months later? I just ordered a VPE for my bathroom.

I needed something with both voice control and a DAC to replace my Google mini there.

I'm planning on mainly single speaker use for there, but joining my existing Airplay 2 group would be nice!

**Reply by OzGav** [reply]
The plan has changed in favour of a dedicated protocol and significant progress has been made. Final ETA is not available sorry.

**Reply by joopdo** [reply]
Seems to be 'resonate' I never heard of Resonate before, curious what are the benefits over Snapcast!

**Reply by OzGav** [reply]
For example: 

- Variable Sampling Rate (each device can sets its own prefered sample rate, bitdepth codec and channels) it wants/supports.

- Remote Control of a Player

- Possibility to stream audio back to server musiccast devices can stream from every device with an aux in to the whole system: Yeah, each device can decide to implement one or more of the roles

- Native support for Announcements

- DSP Support per Player

**Reply by RemakingEden** [reply]
For anyone interested it appears to be this https://github.com/Resonate-Protocol. You can read the spec there. It looks like it has Paulus himself involved which seems promising for it getting implemented. Ive also just read it was mentioned in the latest "state of the open home". I mustve missed that.

Looks exciting @OzGav. The ability to send music back the server is interesting, could that lead to a possible chromecast kind of situation from your phone to all the speakers or is it mainly looking like aux in atm?

**Reply by OzGav** [reply]
Things are changing during development but I think just AUX in at the moment

**Comment by OzGav** [comment]
The Sendspin protocol has been released. Sync of VPE is now possible. You have to use VPE firmware 2025.12.0 or higher

**Comment by joopdo** [comment]
Just want to say, it works really well on first impresssion! On my home speaker I switched from MPD to Sendspin and starting big playlists have become much faster. Will test for TTS as well.
<img width="667" height="500" alt="image" src="https://github.com/user-attachments/assets/cd3b1978-fc63-424b-bf10-7a98ae44aa54" />

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
